csrc/optimization/alias_analysis.cpp

```diff
@@ -247,6 +248,81 @@ void AliasFinder::handle(const LoadStoreOp* permute) {
   analysis_.add(out, in, std::move(out_layout));
 }
 
+void AliasFinder::handle(const SliceOp* slice) {
+  TensorView* in = slice->in();
+  TensorView* out = slice->out();
+
+  const std::vector<IterDomain*>& in_rfactor = in->getMaybeRFactorDomain();
+  const std::vector<IterDomain*>& out_root = out->getRootDomain();
+  const std::vector<IterDomain*>& out_rfactor = out->getMaybeRFactorDomain();
+
```

Comment:
same here, we should check for allocationdomain on outputs. Either give us when we see anything, or double check it after the propagation that the preferred layout is consistent with it.

Reply:
Mark this as resolved and leave the discussion to the other PR.